node_package/src/clientStartup.ts

```diff
@@ -135,13 +135,31 @@ function domNodeIdForEl(el: Element): string {
  * delegates to a renderer registered by the user.
  * @param el
  */
+
+type ComponentCache = { [key: string]: boolean };
+let components: { [key: string]: ComponentCache; } = {};
+
 function render(el: Element, railsContext: RailsContext): void {
   const context = findContext();
   // This must match lib/react_on_rails/helper.rb
   const name = el.getAttribute('data-component-name') || "";
   const domNodeId = domNodeIdForEl(el);
   const props = (el.textContent !== null) ? JSON.parse(el.textContent) : {};
   const trace = el.getAttribute('data-trace') === "true";
+  const path = window.location.pathname;
+  const locationMounted = components[path];
+
+  if (locationMounted) {
+    const componentMounted = locationMounted[props.componentName];
+    if (componentMounted) return;
+
+    components[path] = {
+      ...components[path],
+      [props.componentName]: true,
```

Comment:
@templeman15 Does this mean you can render each component on the given path only once?
Can't imagine you would be rendering same thing multiple times often, but good to watch out for.

Reply:
@tomdracz that's a good point. We don't do that but I definitely see what you're saying. You could technically call react on rails with the same component multiple times on the page..

I was hoping you all would know a unique key that persists across page visits for the rendered components. In our app I see the react dom ID but it changes even on the same component when visiting a cached page.